src/App.jsx

```diff
@@ -14,45 +17,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField, isReverse = false) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReverse ? preparedGoods.reverse() : preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReverse);
+
+  function reset() {
+    setSortField('');
+    setIsReverse(false);
+  }
+
+  return (
+
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn('button is-info',
+              { 'is-light': sortField !== SORT_FIELD_ALPHABET })
+          }
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn('button is-success',
+              { 'is-light': sortField !== SORT_FIELD_LENGTH })
+          }
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn('button is-warning', { 'is-light': !isReverse })
+          }
+          onClick={() => setIsReverse(prev => !prev)}
+        >
+          Reverse
+        </button>
+        {
+          (sortField !== '' || isReverse) && (
```

Comment:
You can write this condition as
```suggestion
          (sortField || isReverse) && (
```